File: src/lib/scala/Slate.Common/src/main/scala/slate/common/app/AppLifeCycle.scala
/**
<slate_header>
  author: Kishore Reddy
  url: https://github.com/kishorereddy/scala-slate
  copyright: 2015 Kishore Reddy
  license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  desc: a scala micro-framework
  usage: Please refer to license on github for more info.
</slate_header>
  */

package slate.common.app

import slate.common.results.ResultFuncs._
import slate.common.Result

trait AppLifeCycle {


  /**
   * initializes the application
    *
   */
  def init(): Result[Boolean] = ok()


  /**
   * executes the application
    *
    * @return
   */
  def exec(): Result[Any] = success("default")


  /**
   * shutdown hook to stop any services
   */
  def end(): Unit = {  }
}

File: src/lib/kotlin/slatekit-common/src/main/kotlin/slatekit/common/app/AppLifeCycle.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.common.app

import slatekit.common.Result
import slatekit.common.results.ResultFuncs.success
import slatekit.common.results.ResultFuncs.yes

interface AppLifeCycle {


    /**
     * initializes the application
     *
     */
    fun init(): Result<Boolean> = yes()


    /**
     * executes the application
     *
     * @return
     */
    fun exec(): Result<Any> = success("")


    /**
     * shutdown hook to stop any services
     */
    fun end(): Unit {}
}
